samples/hello-cube.html

```diff
@@ -0,0 +1,313 @@
+    };
+
+    const depthTexture = device.createTexture(depthTextureDescriptor);
+
+    // GPURenderPassDepthStencilAttachmentDescriptor
+    const depthAttachment = {
+        attachment: depthTexture,
+
+        depthLoadOp = "clear",
+        depthStoreOp = "store",
+        clearDepth: 1.0
+    };
+
+    renderPassDescriptor = {
+        colorAttachments: [colorAttachment],
+        depthStencilAttachment: depthAttachment
+    };
+
+    queue = device.getQueue();
+
+    render();
+}
+
+/* Transform Buffers and Bindings */
+const transformSize = 4 * 16;
+function updateTransformArray(array) {
+    let viewMatrix = mat4.create();
+    mat4.translate(viewMatrix, viewMatrix, vec3.fromValues(0, 0, -5));
+    let now = Date.now() / 1000;
+    mat4.rotate(viewMatrix, viewMatrix, 1, vec3.fromValues(Math.sin(now), Math.cos(now), 0));
+
+    let modelViewProjectionMatrix = mat4.create();
+    mat4.multiply(modelViewProjectionMatrix, projectionMatrix, viewMatrix);
+
+    for (let i = 0; i < 16; i++) {
+        array[i] = modelViewProjectionMatrix[i];
+    }
+}
+
+const transformBufferDescriptor = {
+    size: transformSize,
+    usage: GPUBufferUsage.UNIFORM | GPUBufferUsage.MAP_WRITE
+};
+
+function createTransformBindGroupDescriptor(transformBuffer) {
+    const transformBufferBinding = {
+        buffer: transformBuffer,
+        offset: 0,
+        size: transformSize
+    };
+
+    const transformBufferBindGroupBinding = {
+        binding: 0,
+        resource: transformBufferBinding
+    };
+
+    return {
+        layout: uniformsBindGroupLayout,
+        bindings: [transformBufferBindGroupBinding]
+    };
+}
+
+class MappedBufferBindGroup {
+    constructor(buffer, arrayBuffer, bindGroup) {
+        this.buffer = buffer;
+        this.arrayBuffer = arrayBuffer;
+        this.bindGroup = bindGroup;
+    }
+}
+const mappedGroups = [];
+
+function render() {
+    let mappedGroup;
+
+    if (mappedGroups.length == 0) {
+        // No buffers are mapped and ready for update.
+        const { buffer, arrayBuffer } = device.createMappedBuffer(transformBufferDescriptor);
+        const bindGroup = device.createBindGroup(createTransformBindGroupDescriptor(buffer));
+
+        mappedGroup = new MappedBufferBindGroup(buffer, arrayBuffer, bindGroup);
+    } else
+        mappedGroup = mappedGroups.shift();
+
+    updateTransformArray(mappedGroup.arrayBuffer);
+    mappedGroup.buffer.unmap();
+
+    const commandEncoder = device.createCommandEncoder();
+    renderPassDescriptor.colorAttachments[0].attachment = swapChain.getTexture().createDefaultTextureView();
+    const passEncoder = commandEncoder.beginRenderPass(renderPassDescriptor);
+
+    // Encode drawing commands.
+    passEncoder.setPipeline(pipeline);
+    // Vertex attributes
+    passEncoder.setVertexBuffers(0, [verticesBuffer], [0]);
+    // Bind groups
+    passEncoder.setBindGroup(0, mappedGroup.bindGroup);
+
+    passEncoder.draw(36, 1, 0, 0);
```

Comment:
You should explain where the 36 came from. And probably the 1, 0, 0 part too.

Reply:
Done.